### ISSUE TITLE
setting mac for iOs testing

### ISSUE THREAD

**Issue by michaazran** [appium/sample-code#147]
hi,
i am trying to run appium on my iPhone but i keep getting versions issues
my iPhone is 6S with ios 11.0.3
MAC os is 10.8.5
xCode 5.1.1

when i tried to check the developer options to mark the UIAutomator i got an error from xCode that it is tool old for my iPhone version

i also tried to run the new appium 1.2.7 but i got an error that 10.8.5 is too old and i need at least 10.9
when i ran an old version of appium 1.3.4 it opened successfully
can someone advice what si the shortest way that i can start running appium on that device using my MAC ?

thanks.

**Comment by KazuCocoa** [comment]
It requires Xcode 9, a newer version of Xcode, to build and install an iOS app for iOS11, and Xcode 9 requires the below spec.

> Xcode 9 requires a Mac running macOS 10.12.6 or later.

https://developer.apple.com/library/content/releasenotes/DeveloperTools/RN-Xcode/Chapters/Introduction.html

----

Preparing the following spec is the shortest way to run Appium on your macOS.

- https://github.com/appium/appium/blob/master/docs/en/about-appium/getting-started.md
- https://github.com/appium/appium/blob/master/docs/en/drivers/ios-xcuitest.md#requirements-and-support

**Comment by michaazran** [comment]
thank you for your answer any 
do you have any idea if appium can automeate native application i installed from the AppStore
such as whatsup, facebook, skype ....
applications which i do not have their source code and cannot be opened in Xcode

thanks.

**Comment by KazuCocoa** [comment]
You can automate them, include pre-installed apps, like: https://github.com/appium/appium/issues/9840 . But Appium requires Xcode 8+ and you must upgrade your Xcode first. @michaazran

**Comment by michaazran** [comment]
thanks again
in your link i noticed you described only IOS built in applications , but is it the same for any other?
whatsup, facebook,skype....
is there a way to get the object properties from it using the inspector ?

**Comment by saikrishna321** [comment]
@michaazran you can use AppleScript `osascript -e 'id of app "Facebook"'` or you mobiledevice(https://github.com/imkira/mobiledevice) library can help find all the applications bundleID installed on the device.

**Comment by KazuCocoa** [comment]
@michaazran 

> is it the same for any other?

Yes. Put `bundleId` is significant to manage without installing apps, and how to get the id is https://github.com/appium/sample-code/issues/147#issuecomment-354235646 (thanks!)
